devicetypes/vision-nameseca/zp3111jp-4-in-1-motion.src/zp3111jp-4-in-1-motion.groovy

```diff
@@ -0,0 +1,925 @@
+			0x71: 3,  // Alarm v1 or Notification v4
+			0x72: 2,  // ManufacturerSpecific
+			0x73: 1,  // Powerlevel
+			0x7A: 2,  // FirmwareUpdateMd
+			0x80: 1,  // Battery
+			0x84: 2,  // WakeUp
+			0x85: 2,  // Association
+			0x86: 1,	// Version (2)
+			0x98: 1		// Security
+	]
+}
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	state.checkinInterval = cmd.seconds
+
+	sendUpdatingEvent()
+
+	def msg = "Minimum Check-in Interval is ${cmd.seconds / 60} Minutes"
+	if (cmd.seconds == 600) {
+		log.warn "$msg"
+	}
+	else {
+		logDebug "$msg"
+	}
+
+	runIn(5, finalizeConfiguration)
+
+	return [ createCheckIntervalEvent(cmd.seconds) ]
+}
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	logTrace "WakeUpNotification: $cmd"
+	def cmds = []
+
+	logDebug "Device Woke Up"
+
+	if (getAttrValue("firmwareVersion")) {
+		cmds += configure()
+
+		if (cmds) {
+			cmds << "delay 2000"
+		}
+
+		cmds << wakeUpNoMoreInfoCmd()
+	}
+	else {
+		cmds << versionGetCmd()
+	}
+
+	return cmds ? response(cmds) : []
+}
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+
+	if (val > 100)
+	{
+		val = 100
+	}
+	else if (val < 1)
+	{
+		val = 1
+	}
+
+	state.lastBatteryReport = new Date().time
+	[
+			createEvent(createEventMap("battery", val, "%", true))
+	]
+}
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	logTrace "ManufacturerSpecificReport: ${cmd}"
+	return []
+}
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	logTrace "VersionReport: ${cmd}"
+
+	def version = "${cmd.applicationVersion}.${cmd.applicationSubVersion}"
+	logDebug "Firmware Version: ${version}"
+
+	def result = []
+	if (getAttrValue("firmwareVersion") != "${version}") {
+		result << createEvent(name: "firmwareVersion", value: "${version}", displayed: false)
+	}
+	return result
+}
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	logTrace "ConfigurationReport: ${cmd}"
+	sendUpdatingEvent()
+
+	def val = cmd.scaledConfigurationValue
+
+	def configParam = configParams.find { param ->
+		param.num == cmd.parameterNumber
+	}
+
+	if (configParam) {
+		logDebug "${configParam.name}(#${configParam.num}) = ${val}"
+		state["configVal${cmd.parameterNumber}"] = val
+	}
+	else {
+		logDebug "Parameter ${cmd.parameterNumber} = ${val}"
+	}
+
+	runIn(5, finalizeConfiguration)
+	return []
+}
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	logTrace "BasicReport: $cmd"
+	if (state.refreshAll || !device.currentValue("motion")) {
+		sendMotionEvents(cmd.value)
+	}
+	return []
+}
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	logTrace "BasicSet: $cmd"
+	sendMotionEvents(cmd.value)
+	return []
+}
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
```

Comment:
Is this device actually using Notification V3 or Alarm v2?

Reply:
It uses Notification V3, thank you.